FWCore/Services/plugins/CondorDynamicResize.cc

```diff
@@ -0,0 +1,340 @@
+#include <cassert>
+#include <cctype>
+#include <chrono>
+#include <iostream>
+#include <fstream>
+#include <future>
+#include <sstream>
+#include <thread>
+#include <vector>
+
+namespace edm {
+
+  namespace service {
+
+    class CondorDynamicResizeService
+    {
+
+    public:
+
+      explicit CondorDynamicResizeService(ParameterSet const& pset,
+                                          edm::ActivityRegistry& ar);
+      ~CondorDynamicResizeService() {}
+      CondorDynamicResizeService(const CondorDynamicResizeService&) = delete;
+      CondorDynamicResizeService& operator=(const CondorDynamicResizeService&)
+        = delete;
+
+      static void fillDescriptions(ConfigurationDescriptions &descriptions);
+
+    private:
+
+      // Main event loop for the management thread.
+      void manageTasks(std::future<int> stop_signal);
+      static void manageTasksHelper(CondorDynamicResizeService *svc,
+                                    std::future<int> stop_signal);
+
+      // Parse the HTCondor environment and determine the optimal number
+      // of cores to utilise.  Returns 0 on error.  As m_max_cores can change
+      // during the process life, getTargetCores may be larger than max cores.
+      unsigned int getTargetCores() const;
+
+      // Signal from CMSSW that we should shutdown all our threads and tasks.
+      void shutdown();
+
+      bool m_debug = false;
+
+      // Indicates management thread should halt all futures and shutdown.
+      std::promise<int> m_stop_signal;
+
+      // Management of thread counts.  This is an atomic because this will
+      // be read from the helper thread (m_mgmt_thread) and written to in
+      // a callback invoked by the framework.
+      std::atomic<unsigned int> m_max_cores;
+
+      // Management thread
+      std::thread m_mgmt_thread;
+    };
+
+  }
+
+}
+
+namespace {
+
+  void trim(std::string& s, const std::string& drop = " \t") {
+    std::string::size_type p = s.find_last_not_of(drop);
+    if (p != std::string::npos) {
+          s = s.erase(p+1);
+    }
+    s = s.erase(0, s.find_first_not_of(drop));
+  }
+
+
+  class SleeperTask : public tbb::task
+  {
+
+  public:
+
+    explicit SleeperTask(std::future<int> halt_signal, bool debug) :
+      m_debug(debug),
+      m_halt_signal(std::move(halt_signal)) {}
+
+    tbb::task* execute() override {
+      if (m_debug) {std::cout << "Launching a TBB-based sleep task.\n";}
+      m_halt_signal.wait();
+      if (m_debug) {std::cout << "Exiting from a TBB-based sleep task.\n";}
+      return nullptr;
+    }
+
+  private:
+
+    bool m_debug = false;
+    std::future<int> m_halt_signal;
+  };
+
+}
+
+using namespace edm::service;
+
+CondorDynamicResizeService::CondorDynamicResizeService(
+                           ParameterSet const& pset,
+                           edm::ActivityRegistry& ar) :
+  m_debug(false),
+  m_max_cores(1)
+{
+  if (pset.existsAs<bool>("debug"))
+  {
+    m_debug = pset.getUntrackedParameter<bool>("debug");
```

Comment:
Move this to the initialization. There is no need to check `existAs` since the `fillDescriptions` will make sure the value is set.